### PR TITLE
[FEATURE][REFACTOR] MRMFeatureQC: Additional QC criteria for component groups

### PR DESCRIPTION
**Pull request by pasqoo**
The following QC criteria have been added to `MRMFeatureQC::ComponentGroupQCs`:
```
double retention_time_l;
double retention_time_u;
double intensity_l;
double intensity_u;
double overall_quality_l;
double overall_quality_u;
std::map<String,std::pair<double,double>> meta_value_qc;
```
The change is accompanied by an update of _MRMFeatureQCFile_ to support said addition.
Two files are now expected when loading components and components groups QC criteria (the behavior is controlled through the flag `is_component_group` of `load()`).

### CODE REVIEW

**Comment by pasqoo** [comment]
travis timed ou

File: src/openms/include/OpenMS/ANALYSIS/OPENSWATH/MRMFeatureQC.h [hunk 1]
```diff
@@ -114,28 +114,42 @@ namespace OpenMS
...
+      /// retention time upper bound
+      double retention_time_u;
+      /// intensity lower bound
+      double intensity_l;
+      /// intensity upper bound
+      double intensity_u;
+      /// overall quality lower bound
+      double overall_quality_l;
+      /// overall quality upper bound
+      double overall_quality_u;
```
Comment (hroest) [inline src/openms/include/OpenMS/ANALYSIS/OPENSWATH/MRMFeatureQC.h]: probably you want to add those as well to the pxd file to expose them to Python

File: src/openms/include/OpenMS/ANALYSIS/OPENSWATH/MRMFeatureQC.h [hunk 2]
```diff
@@ -169,7 +183,7 @@ namespace OpenMS
     /// list of all component group QCs
     std::vector<ComponentGroupQCs> component_group_qcs;
     /// list of all component group pair QCs
-    std::vector<ComponentGroupQCs> component_group_pair_qcs;
+    std::vector<ComponentGroupPairQCs> component_group_pair_qcs;
```
Comment (hroest) [inline src/openms/include/OpenMS/ANALYSIS/OPENSWATH/MRMFeatureQC.h]: also change this in PyOpenMS

**Comment by hroest** [comment]
pyOpenMS has some errors that need fixing: http://cdash.openms.de/viewBuildError.php?buildid=260440

**Comment by pasqoo** [comment]
@hroest Thanks! Files updated. Let's see what the tests will say :smile:

**Comment by pasqoo** [comment]
Travis timed out on tests, but the other builds seem to look fine:
http://cdash.openms.de/buildSummary.php?buildid=260832
http://cdash.openms.de/buildSummary.php?buildid=260837

**Comment by hroest** [comment]
ok I restarted the tests

File: src/pyOpenMS/pxds/MRMFeatureQC.pxd [hunk 3]
```diff
@@ -58,5 +66,5 @@ cdef extern from "<OpenMS/ANALYSIS/OPENSWATH/MRMFeatureQC.h>" namespace "OpenMS"
         
         libcpp_vector[MRMFQC_ComponentQCs] component_qcs
         libcpp_vector[MRMFQC_ComponentGroupQCs] component_group_qcs
-        libcpp_vector[MRMFQC_ComponentGroupQCs] component_group_pair_qcs
+        libcpp_vector[MRMFQC_ComponentGroupPairQCs] component_group_pair_qcs
```
Comment (hroest) [inline src/pyOpenMS/pxds/MRMFeatureQC.pxd]: I dont think `ComponentGroupPairQCs` is defined anywhere, did this compile on your machine?

**Comment by pasqoo** [comment]
Another timeout, unfortunately

**Comment by pasqoo** [comment]
@hroest PR is up to date :smile:

**Pull request merged**